### ISSUE TITLE
Rename Env Keys

### ISSUE THREAD

**Issue by Lypsolon** [ynput/ayon-cpp-api#6]
# Tasks 

- [x] Switch to the New Logger from Cpp Dev Tools 
- [x] Remove the Env Keys and make them Required at Cpp Api Construction(so the caller needs to implement the Env Key setup).
- [ ] [Change the Ayon Usd Resolver to setup the New Ayon Cpp Api](https://github.com/ynput/ayon-usd-resolver/issues/37)
- [ ] [Change Ayon Usd addon Resolver Env variables to be used](https://github.com/ynput/ayon-usd/issues/67)

The current name for the logger settings could be nicer. Let's change them to be a bit more descriptive.
 
AYONLOGGERLOGLVL = LOG_LEVEL
AYONLOGGERFILELOGGING = ENABLE_FILE_LOGGING
AYONLOGGERFILEPOS = LOG_FILE_PATH
 
[Blocking PR]
- [ ] https://github.com/ynput/ayon-cpp-api/pull/27

[PRs Waiting on this PR]
- [ ] https://github.com/ynput/ayon-usd-resolver/issues/37

**Comment by Lypsolon** [comment]
@antirotor @BigRoy. 

The CppApi is now changed to use no Env Vrabiables. 
this enables us to change the env keys from the Resolver. 

but there are a few Questions about what names we want them to have. 
1. i believe we should keep the AYON_ prefix as LOG_LEVEL dose not inform us what we are controlling 
2. do we want to rename other Env keys while we are on it ? 
```
AYONLOGGERLOGLVL -> AYON_USD_RESOLVER_LOGLVL or AYON_USD_RESOLVER_LOG_LVL
AYONLOGGERFILELOGGING -> AYON_USD_RESOLVER_FILELOGGING 
AYONLOGGERFILEPOS -> AYON_USD_RESOLVER_LOG_FILE
AYON_LOGGIN_LOGGIN_KEYS -> AYON_USD_RESOLVER_LOGGIN_KEYS

ENABLE_STATIC_GLOBAL_CACHE -> AYON_USD_RESOLVER_ENABLE_PINNING
PINNING_FILE_PATH -> AYON_USD_RESOLVER_PINNING_FILE
PROJECT_ROOTS -> AYON_USD_RESOLVER_PROJECT_ROOTS

AYON_API_KEY -> cant be renamed
AYON_SERVER_URL -> cant be renamed
AYON_SITE_ID -> cant be renamed
AYON_PROJECT_NAME -> cant be renamed

TF_DEBUG -> cant be renamed
```

**Comment by Lypsolon** [comment]
Also i believe this should be a major version change
or 
somehow show in the version its backwards incomparable. 

Reason: 
if some one has build something ontop of the resolver this will be a braking change. 
also the addons should be informed that there is a braking change 
(currently only we are working on the addons that consume the Resolver but it might be a good moment to follow the idea of big version ups on backward incompatible changes ?)

**Comment by Lypsolon** [comment]
> Why do we need this separately?
> 
> ```
> AYONLOGGERFILELOGGING -> AYON_USD_RESOLVER_FILELOGGING 
> AYONLOGGERFILEPOS -> AYON_USD_RESOLVER_LOG_FILE
> ```
> 
> Why can't we just assume that when `AYON_USD_RESOLVER_LOG_FILE` is set that we want file logging enabled? 

we dont do this for developer convenience. this way you can have a file set and disable it. eg for testing or for re running a job without it. 
its a handy little feature kinda like the enabled bool on an pyblish instance. 

also remember if you set an env variable to be empty on a good few Linux systems the env variable will not be gone. 
so you might end up with checking if its removed. (its a rare game but cmake hits me with it every now and then)

> Preferably we just keep the `AYON_USD_RESOLVER_LOG_*` keys so they are 'together'.

agreed. 
like this ? 
```
AYONLOGGERLOGLVL -> AYON_USD_RESOLVER_LOG_LVL
AYONLOGGERFILELOGGING -> AYON_USD_RESOLVER_LOG_FILELOGGING 
AYONLOGGERFILEPOS -> AYON_USD_RESOLVER_LOG_FILE
AYON_LOGGIN_LOGGIN_KEYS -> AYON_USD_RESOLVER_LOG_KEYS

ENABLE_STATIC_GLOBAL_CACHE -> AYON_USD_RESOLVER_ENABLE_PINNING
PINNING_FILE_PATH -> AYON_USD_RESOLVER_PINNING_FILE
PROJECT_ROOTS -> AYON_USD_RESOLVER_PINNING_ROOTS

AYON_API_KEY -> cant be renamed
AYON_SERVER_URL -> cant be renamed
AYON_SITE_ID -> cant be renamed
AYON_PROJECT_NAME -> cant be renamed

TF_DEBUG -> cant be renamed
```

> And for this one:
> 
> ```
> AYON_USD_RESOLVER_PROJECT_ROOTS
> ```
> 
> I have the tendency to also change that around so it's related to the pinning. So:
> 
> ```
> AYON_USD_RESOLVER_PROJECT_ROOTS -> AYON_USD_RESOLVER_PINNING_PROJECT_ROOTS or AYON_USD_RESOLVER_PINNING_ROOTS
> ```

i like `AYON_USD_RESOLVER_PINNING_ROOTS` it sounds nice, no other reason tbh ;-)

**Comment by BigRoy** [comment]
> we dont do this for developer convenience. this way you can have a file set and disable it. eg for testing or for re running a job without it.
> its a handy little feature kinda like the enabled bool on an pyblish instance.

Fine with me - bit verbose but hey - if there's a use case, fine with me. @antirotor thoughts about having both enabled?
```
AYON_USD_RESOLVER_FILELOGGING 
AYON_USD_RESOLVER_LOG_FILE
```
Or maybe a better name is:
```
AYON_USD_RESOLVER_LOG_FILE_ENABLED
AYON_USD_RESOLVER_LOG_FILE
```

**Comment by Lypsolon** [comment]
> ```
> AYON_USD_RESOLVER_FILELOGGING 
> AYON_USD_RESOLVER_LOG_FILE
> ```
> 
> Or maybe a better name is:
> 
> ```
> AYON_USD_RESOLVER_LOG_FILE_ENABLED
> AYON_USD_RESOLVER_LOG_FILE
> ```

i vote for the second option they are more direct and you don't need to guess.

**Comment by Lypsolon** [comment]
[Status]
this message will give a simple update on the situation. 
this task is still blocked by this PR https://github.com/ynput/ayon-cpp-api/pull/27 
im would like to know who i could assign maybe some one with some c++ experience that would be nice. 
in the end of the day everything in the other PR is a implementation question.

**Comment by antirotor** [comment]
I vote for the second option too btw.

**Comment by dee-ynput** [comment]
#27 is ready to merge, merge it and go on with this one.
Just keep in mind this has a low priority so focus on it once you have availability 👍

**Comment by Lypsolon** [comment]
the fix for this will be implemented inside ayon usd resolver. 
we are now at the point where the cpp api dose not need env variables any more.

**Comment by mkolar** [comment]
@antirotor @BigRoy @dee-ynput do any of you know whether we can close this, considering the last comment from Lyon?

**Comment by BigRoy** [comment]
Looking at the PR's description and individual 'issues' that are not checked in those bullet point lists I assume there's still stuff open- but I've no idea myself unfortunately. I wouldn't know where to look. 🗡

**Comment by dee-ynput** [comment]
I believe this one is obsolete.
But I wouldn't bet an arm on it 😬

**Comment by tadeas-hejnic** [comment]
As far as I know this should be done in resolver as the Cpp Api doesn't work with env variables anymore, but in resolver helper function that handles the env load are still the old names. [PR](https://github.com/ynput/ayon-usd-resolver/pull/46) that is not merged yet...

**Comment by tadeas-hejnic** [comment]
So in my opinion, it should be closed in this repository as cpp api doesn't work with env variables anymore..

**Comment by BigRoy** [comment]
I'd say, feel free to close - and maybe comment on the other repo where it may still be relevant to 'revive' that to make sure it's clear that _there_ it's still relevant.

**Comment by Lypsolon** [comment]
Jup This issue was suposed to be closed when the PR on the Revolver is Merged so you might as well do so now. 

[Edit] if you close this it might be Good to open an Issue on usd addon to ensure to Rename the variabels when you merge the stuff in the resolver as they arent the same any More.
